src/cmap/connect.ts

```diff
@@ -61,7 +60,16 @@ export function connect(options: ConnectionOptions, callback: Callback<Connectio
     if (options.autoEncrypter) {
```

Comment:
Just because we've come this far, do you want to refactor up one more function?

```typescript
const makeConnectionAsync = promisify(makeConnection);

async function connectAsync(..) {


}
export const connect = callbackify(connectAsync);
```

I don't think it makes sense to make `makeConnection` async, because ideally we'll refactor it eventually so it's not performing io at all (purely synchronous).  It also currently handles event listeners on the socket, so just wrapping it in a promisify seems more straightforward.

With this last refactor, the auth module is basically completely refactored.

Reply:
I looked into making this change but it would only be usable here, the socks5 logic would need to be untangled from callbacks as well. Or we leave socks5 unchanged and it still uses the callback form at which point we haven't got much value out of making a promisified version. Let's stick with what we got so far